lib/shopify_cli/theme/dev_server/hot-reload.js

```diff
@@ -17,18 +17,27 @@
 
   connect();
 
+  let nonDynamicFileChanged = false;
```

Comment:
I think this should be inside the `if`? Or else it will never be reseted to `false`.

EDIT: However, I think because the page is reloaded it might not be a problem. But still I would move inside the `if` to make this clearer.